language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingConstructorExpressionNodeContext.java

```diff
@@ -210,6 +207,45 @@ private boolean withinComputedNameContext(BallerinaCompletionContext context, No
         return Optional.empty();
     }
 
+    private Optional<RecordTypeSymbol> getRecordForAssignment(AssignmentStatementNode assignmentNode,
+                                                              CompletionContext ctx) {
+        Node varRef = assignmentNode.varRef();
+        String varName;
+        if (varRef.kind() == SyntaxKind.INDEXED_EXPRESSION
+                && ((IndexedExpressionNode) varRef).containerExpression().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+            varName = ((SimpleNameReferenceNode) ((IndexedExpressionNode) varRef).containerExpression()).name().text();
+        } else if (varRef.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+            varName = ((SimpleNameReferenceNode) varRef).name().text();
+        } else {
+            return Optional.empty();
+        }
+
+        List<Symbol> visibleSymbols = ctx.visibleSymbols(ctx.getCursorPosition());
```

Comment:
Wouldn't `symbol()` do for this?

Reply:
The cursor position is the position where the completion has been triggered and we need to capture the LHS symbol, which should be handled with syntax tree visitor. Syntax tree visitor based implementation should be bottom up and need a revamping in the current implementation. Which is planned for the next iteration.